src/Microsoft.IdentityModel.Clients.ActiveDirectory/Core/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -1,131 +1,138 @@
-        {
-            Environment = environment;
-            ClientId = clientId;
-            TokenType = tokenType;
-            NormalizedScopes = scopes;
-            TenantId = tenantId;
-            Secret = secret;
-            ExpiresOnUnixTimestamp = CoreHelpers.DateTimeToUnixTimestamp(accessTokenExpiresOn);
-            ExtendedExpiresOnUnixTimestamp = CoreHelpers.DateTimeToUnixTimestamp(accessTokenExtendedExpiresOn);
-            CachedAt = CoreHelpers.CurrDateTimeInUnixTimestamp();
-            RawClientInfo = rawClientInfo;
-
-            InitUserIdentifier();
-        }
-
-        [DataMember(Name = "realm")]
         internal string TenantId { get; set; }
 
         /// <summary>
         /// String comprised of scopes that have been lowercased and ordered.
         /// </summary>
         /// <remarks>Normalization is important when creating unique keys.</remarks>
-        [DataMember(Name = "target", IsRequired = true)]
         internal string NormalizedScopes { get; set; }
-
-        [DataMember(Name = "cached_at", IsRequired = true)]
         internal string CachedAt { get; set; }
-
-        [DataMember(Name = "expires_on", IsRequired = true)]
         internal string ExpiresOnUnixTimestamp { get; set; }
-
-        [DataMember(Name = "ext_expires_on")]
         internal string ExtendedExpiresOnUnixTimestamp { get; set; }
-
-        [DataMember(Name = "user_assertion_hash", EmitDefaultValue = false)]
         public string UserAssertionHash { get; set; }
 
-        [DataMember(Name = "access_token_type")]
-        internal string TokenType { get; set; }
+        internal string Authority =>
+            string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", Environment, TenantId ?? "common");
 
-        internal string Authority => string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", Environment, TenantId ?? "common");
         internal SortedSet<string> ScopeSet => ScopeHelper.ConvertStringToLowercaseSortedSet(NormalizedScopes);
 
-        internal DateTimeOffset ExpiresOn
+        internal DateTimeOffset ExpiresOn => CoreHelpers.UnixTimestampStringToDateTime(ExpiresOnUnixTimestamp);
+        internal DateTimeOffset ExtendedExpiresOn => CoreHelpers.UnixTimestampStringToDateTime(ExtendedExpiresOnUnixTimestamp);
+        public bool IsExtendedLifeTimeToken { get; set; }
+
+        internal static MsalAccessTokenCacheItem FromJsonString(string json)
+        {
+            return FromJObject(JObject.Parse(json));
+        }
+
+        internal static MsalAccessTokenCacheItem FromJObject(JObject j)
         {
-            get
+            long cachedAt = JsonUtils.ExtractParsedIntOrZero(j, StorageJsonKeys.CachedAt);
+            long expiresOn = JsonUtils.ExtractParsedIntOrZero(j, StorageJsonKeys.ExpiresOn);
+
+            // This handles a bug with the name in previous MSAL.  It used "ext_expires_on" instead of
+            // "extended_expires_on" per spec, so this works around that.
```

Comment:
could you shorten the comment and maybe add a link to a github issue or the wiki instead?

Reply:
given our stance on v2/v3 compat and this being an internal only used field, i've just removed it.